### ISSUE TITLE
Allow us to silence type alias related runtime errors

### ISSUE THREAD

**Issue by AlexEvanczuk** [sorbet/sorbet#2673]
### Problem

https://sorbet.run/#require%20%27sorbet-runtime%27%0A%0Amodule%20Contracts%0A%20%20class%20Maybe%0A%20%20end%0Aend%0A%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20X%20%3D%20Contracts%3A%3AMaybe.new%0A%0A%20%20sig%20%7Breturns(X)%7D%0A%20%20def%20self.foo%0A%20%20end%0Aend%0A%0AA.foo

I'd like to use failure overrides to prevent this from erroring in production when this code is executed.

### Proposed solution
I'd like this to be caught by `inline_type_error_handler`

**Comment by jez** [comment]
I don't think this should piggy back on inline_type_error_handler, but i'm fine either using something like `sig_builder_error_handler` or introducing a new error handler for `T::Utils.coerce`

**Comment by jez** [comment]
More generally we might want to audit all uses of `raise` in `sorbet-runtime` and replace them with calls to `T::Configuration` error handlers, or ensure that if they're using `raise` that they're raising a specific exception class (i.e., no RuntimeError / `raise "foo"`) and that that specific error class is rescued by calling a `T::Configuration` handler.

**Comment by AlexEvanczuk** [comment]
Is this another instance of the same issue?

I have the following issue which caused some production paths to be broken despite the runtime error overrides:
```
Incompatible return type in override of method `METHOD_NAME`:\n* Base: `String` (in #\u003cClass:BASE_CLASS\u003e at /var/www/app/services/path_to_base_class.rb:13)\n* Override: `NilClass` (in #\u003cClass:CHILD_CLASS\u003e at /var/www/app/services/path_to_child_class.rb:19)\n(The types must be covariant.)
```
